providers/universal-provider/src/utils/misc.ts

```diff
@@ -101,3 +101,29 @@ export function normalizeNamespaces(namespaces: NamespaceConfig): NamespaceConfi
 export function parseCaip10Account(caip10Account: string): string {
   return caip10Account.includes(":") ? caip10Account.split(":")[2] : caip10Account;
 }
+
+/**
+ * Populates the chains array for each namespace with the chains extracted from the accounts if are otherwise missing
+ */
+export function populateNamespacesChains(
+  namespaces: SessionTypes.Namespaces,
+): Record<string, SessionTypes.Namespace> {
+  const parsedNamespaces: Record<string, SessionTypes.Namespace> = {};
+  for (const [key, values] of Object.entries(namespaces)) {
+    const methods = values.methods || [];
+    const events = values.events || [];
+    const accounts = values.accounts || [];
+    const chains = isCaipNamespace(key)
```

Comment:
The shorthand via `isCaipNamespace` here is a bit confusing at first, since the intent is actually to see if the key is a full namespace + chainId definition right (e.g. `eip155:5`)?

```suggestion
    // If the key includes a CAIP separator `:` we know it's a namespace + chainId (e.g. `eip155:1`)
    const chains = isCaipNamespace(key)
```